File: src/cpp/lib/student.cpp
#include <iostream>
#include <string>

using namespace std;

class Student : public Person
{
    protected:
    string institution;
    string major;
    static int count;
    public:
    Student()
    {
        count++;
    }
    void setdata(string fname, string lname, string g, int a, float h, float w, string i, string m)
    {
        first_name = fname;
        last_name = lname;
        gender = g;
        age = a;
        height = h;
        institution = i;
        major = m;
    }
    void displaydata()
    {
        cout<<institution<<endl;
        cout<<major<<endl;
    }
    void talk()
    {
        cout<<"Hello! I am "<<first_name<<" "<<last_name<<". I am a student at "<<institution<<", studying "
        <<major<<"."<<endl;
    }
    static int studtotal()
    {
        return count;
    }
};

File: src/python/lib/student.py
from lib.person import Person
from dataclasses import dataclass

@dataclass
class Student(Person):


    institution: str
    major: str



    def talk(self):
        

        print("Hello! I am {} {} . I am a student at {}, studying {}.".format(self.first_name, self.last_name, self.institution, self.major))


    